ext/node/ops/sqlite/statement.rs

```diff
@@ -543,20 +543,38 @@ impl StatementSync {
         anon_start += 1;
       }
 
-      let mut anon_idx = 1;
+      // SAFETY: `raw` is a valid pointer to a sqlite3_stmt.
+      let sql_param_count = unsafe { ffi::sqlite3_bind_parameter_count(raw) };
+
+      let mut positional_idx = 1;
       for i in anon_start..params.length() {
-        // SAFETY: `raw` is a valid pointer to a sqlite3_stmt.
-        while !unsafe { ffi::sqlite3_bind_parameter_name(raw, anon_idx) }
-          .is_null()
-        {
-          anon_idx += 1;
+        // Find the next positional parameter slot.
+        // Skip named parameters (:name, $name, @name) but include anonymous (?)
+        // and numbered (?NNN) parameters.
+        while positional_idx <= sql_param_count {
+          // SAFETY: `raw` is a valid pointer to a sqlite3_stmt.
+          let name_ptr =
+            unsafe { ffi::sqlite3_bind_parameter_name(raw, positional_idx) };
+          if name_ptr.is_null() {
+            // Anonymous parameter - use this slot
+            break;
+          }
+          // Check if it's a numbered parameter (starts with '?')
+          // SAFETY: name_ptr is not null, checked above.
+          let first_byte = unsafe { *name_ptr as u8 };
+          if first_byte == b'?' {
+            // Numbered parameter like ?1, ?2 - use this slot
+            break;
+          }
```

Comment:
nit, can be single if:
```suggestion
          if name_ptr.is_null()
              // SAFETY: short-circuiting guarantees name_ptr is non-null here
              || unsafe { *name_ptr as u8 == b'?' }
          {
              break;
          }
```